lib/matplotlib/image.py

```diff
@@ -969,7 +973,21 @@ def set_extent(self, extent):
         state is not changed, so following this with ``ax.autoscale_view()``
         will redo the autoscaling in accord with ``dataLim``.
         """
-        self._extent = xmin, xmax, ymin, ymax = extent
+        (xmin, xmax), (ymin, ymax) = self.axes._process_unit_info(
+            [("x", [extent[0], extent[1]]),
+             ("y", [extent[2], extent[3]])],
+            kwargs)
```

Comment:
```suggestion
            kwargs)
        if len(kwargs):
            raise ValueError(f"set_extent did not consume all of the kwargs passed {list(kwargs)!r} were unused")
```
I `_process_unit_info` is very forgiving and is meant to be used in contexts where something else is going to make sure we do not have any extra kwargs.  However in this case we are silently drop them on the floor (which in not great because now a typo will get silently ignored _and_ not do what the wants....bugs where I am sure that `xnuints` is right but not doing what I want are among the most maddening!) 

Because the keywords used depends on the type of axes we are sitting on (😱 ) I think that this is the best option?

Reply:
made the changes

Reply:
The build is green, could you please have a look?

Reply:
I don't see any check for leftover `kwargs`?

Reply:
I had added a check for the leftover kwargs. Could you please let me know what is supposed to be done when there are kwargs remaining? For eg, if we raise a ValueError as before, then other test cases break - like when `cmap` is used as a kwarg

Reply:
You will need to be more explicit about what gets passed to `set_extent` from `__init__`, so that you can raise in the manner noted above.

Reply:
done. Does imshow even need to pass kwargs to set_extents? I have removed it. So that the user can pass those keyword arguments by calling set_extents and not in imshow. Does that sound correct?